plugins/modules/katello_content_view_filter.py

```diff
@@ -252,51 +234,63 @@ def main():
+    changed, content_view_filter = module.ensure_entity('content_view_filters', entity_dict, content_view_filter, params=cv_scope,
+                                                        state=filter_state, entity_spec=content_filter_spec)
+
+    if content_view_filter is not None:
+        cv_filter_scope = {'content_view_filter_id': content_view_filter['id']}
         if 'errata_id' in entity_dict:
-            rule_map = content_filter_rule_erratum_id_map
-            entity_dict['errata'] = find_errata(module, id=entity_dict['errata_id'], organization=organization)
-            content_view_filter_rule_entity = find_content_view_filter_rule(module, content_view_filter=entity_dict['content_view_filter'],
-                                                                            errata=entity_dict['errata'], failsafe=True)
+            # should we try to find the errata the user is asking for? or just pass it blindly?
+            # errata = module.find_resource('errata', 'id={0}'.format(entity_dict['errata_id']), params=scope)
+            rule_spec = content_filter_rule_erratum_id_spec
+            search_scope = {'errata_id': entity_dict['errata_id']}
+            search_scope.update(cv_filter_scope)
+            search = None
         else:
-            rule_map = globals()['content_filter_rule_%s_map' % (entity_dict['filter_type'])]
-            content_view_filter_rule_entity = find_content_view_filter_rule(module, content_view_filter=entity_dict['content_view_filter'],
-                                                                            name=entity_dict['rule_name'], failsafe=True)
+            rule_spec = globals()['content_filter_rule_%s_spec' % (entity_dict['filter_type'])]
+            search_scope = cv_filter_scope
+            if entity_dict['rule_name'] is not None:
+                search = 'name="{0}"'.format(entity_dict['rule_name'])
+            else:
+                search = None
+        # not using find_resource_by_name here, because not all filters (errata) have names
+        content_view_filter_rule = module.find_resource('content_view_filter_rules', search, params=search_scope, failsafe=True)
 
         if entity_dict['filter_type'] == 'package_group':
-            entity_dict['uuid'] = find_package_group(module, name=entity_dict['rule_name']).uuid
+            package_group = module.find_resource_by_name('package_groups', entity_dict['rule_name'], params=scope)
+            entity_dict['uuid'] = package_group['uuid']
+
+        # drop 'name' from the dict, as otherwise it might override 'rule_name'
+        rule_dict = entity_dict.copy()
+        rule_dict.pop('name', None)
 
-        content_view_filter_rule_dict = sanitize_entity_dict(entity_dict, rule_map)
         check_missing = ['min_version', 'max_version', 'version', 'start_date', 'end_date', 'architecture', 'date_type']
```

Comment:
this is not used anymore and I don't completely understand @sean797's old comment in the cement code:

```
# check_missing is a special case, Foreman sometimes returns different values
# depending on what 'type' of same object you are requesting. Content View
# Filters are a prime example. We list these attributes in `check_missing`
# so we can ensure the entity is as the user specified.
```

Reply:
I also am not sure, but i know, that we switched the searching for properties to compare from the (server responded) object to the (user provided) dictionary of desired values. This made the update method look so much simpler. Maybe it fixed this one also. Can we rely on the existing tests here?

Reply:
the test do fail on `master` when I remove the `check_missing` code:

```diff
diff --git a/plugins/module_utils/ansible_nailgun_cement.py b/plugins/module_utils/ansible_nailgun_cement.py
index 162ff3b..cb8ed46 100644
--- a/plugins/module_utils/ansible_nailgun_cement.py
+++ b/plugins/module_utils/ansible_nailgun_cement.py
@@ -292,21 +292,9 @@ def update_entity(old_entity, entity_dict, module, check_missing, check_type, fo
                 if not fields_equal(value, entity_dict[key]):
                     volatile_entity.__setattr__(key, entity_dict[key])
                     fields.append(key)
-            # check_missing is a special case, Foreman sometimes returns different values
-            # depending on what 'type' of same object you are requesting. Content View
-            # Filters are a prime example. We list these attributes in `check_missing`
-            # so we can ensure the entity is as the user specified.
-            if check_missing is not None and key not in entity_dict and key in check_missing:
-                volatile_entity.__setattr__(key, None)
-                fields.append(key)
         if force_update:
             for key in force_update:
                 fields.append(key)
-        if check_missing is not None:
-            for key in check_missing:
-                if key in entity_dict and key not in volatile_entity.get_values():
-                    volatile_entity.__setattr__(key, entity_dict[key])
-                    fields.append(key)
         if len(fields) > 0:
             if not module.check_mode:
                 result = volatile_entity.update(fields)
```

So "maybe"? :)

Reply:
OK for me, but i'd say it _is_ @sean797 s call.

Reply:
>  we switched the searching for properties to compare from the (server responded) object to the (user provided) dictionary of desired values. This made the update method look so much simpler. Maybe it fixed this one also.

I think that probably removed any need for `check_missing` :+1:

Reply:
Yes - we can rely on tests for this, previously without it most filter types would always report back as `changed`.
So this can be removed.